### ISSUE TITLE
[Feature] Detect invalid inputs

### ISSUE THREAD

**Issue by eifinger** [rhysd/actionlint#336]
I have several workflows which can be triggered manually (`workflow_dispatch`) and by other workflows (`workflow_call`). Usually those take the same inputs and sometimes I forget to declare the inputs for both triggers.

This leads to errors like `Invalid input, missingInput is not defined in the referenced workflow.`

I created a sample project to show this.
[This workflow](https://github.com/eifinger/actionlint-playground/blob/main/.github/workflows/this-calls.yml) calls [this workflow](https://github.com/eifinger/actionlint-playground/blob/main/.github/workflows/this-gets-called.yml) with `missingInput`.
As it can be seen in [this run](https://github.com/eifinger/actionlint-playground/actions/runs/5818017838) this is invalid as `missingInput` is not defined as an input for `workflow_call`.

It would be great if actionlint would detect that.

Constraint: As checking remote version input would include to download and parse the remote workflows I would restrict this feature for workflows of the same repository.

**Comment by eifinger** [comment]
Some extension in vscode detects the problem. I have to find out which one it is
<img width="709" alt="image" src="https://github.com/rhysd/actionlint/assets/1481961/aceb5960-2988-4953-ae2f-f5d20e5b377c">

**Comment by EwiththeBowtie** [comment]
I agree this would be very handy and I'm willing to take it on as this is the most common issue in workflows that is not caught by actionlint on my current development team.  Has anyone else started work on this?
